x-pack/plugins/data_enhanced/public/autocomplete/providers/kql_query_suggestion/value.ts

```diff
@@ -23,29 +25,27 @@ const wrapAsSuggestions = (start: number, end: number, query: string, values: st
       end,
     }));
 
-export const setupGetValueSuggestions: KqlQuerySuggestionProvider = (core) => {
+export const setupGetValueSuggestions: KqlQuerySuggestionProvider = () => {
   return async (
     { indexPatterns, boolFilter, signal },
     { start, end, prefix, suffix, fieldName, nestedPath }
   ): Promise<QuerySuggestion[]> => {
-    const allFields = flatten(
-      indexPatterns.map((indexPattern) =>
-        indexPattern.fields.map((field) => ({
-          ...field,
-          indexPattern,
-        }))
-      )
-    );
-
     const fullFieldName = nestedPath ? `${nestedPath}.${fieldName}` : fieldName;
-    const fields = allFields.filter((field) => field.name === fullFieldName);
+
+    const indexPatternFieldEntries: Array<[IIndexPattern, IFieldType]> = [];
```

Comment:
Just want to make sure these are the correct types. They're necessary if someone is passing in js objects instead of classes instances but otherwise we should use `IndexPattern` and `IndexPatternField`

Reply:
If we could add that in a follow-up PR I think that'd be ideal since it will also include some refactoring of all the suggestors (value, conjunction, fields, etc.)